app/views/hackathons/submission_mailer/approval.html.erb

```diff
@@ -0,0 +1,4 @@
+<p>
+  Your hackathon ("<%= @hackathon.name %>") has been approved!
+  Check it out at <%= link_to "hackathons.hackclub.com", Hackathons::WEBSITE %>.
```

Comment:
just FYI, it can take the hackathon site a while to rebuild and show new results. So a user may click the link immediately after delivery and not see their hackathon posted

Reply:
How long? I thought the API was cached by the frontend, but also called every time (and its data used to update the page). I was thinking it might be nice to have a small grace period, like 5-15 minutes, to handle accidental button presses and whatnot, so maybe that could solve this as well.

Reply:
I'm not sure how long the delay is, but 5-15 mins sounds about right.

As for accidental button presses, we could look into debouncing approval status changes: Don't send email and expose via API until it's been approved for at least 1 min.

- For the API, we can do something like `where("approved_at <= ?", 1.min.ago)`
- For the email, we can delay the job by 1 min, and add a condition at the start of the job to cancel (return early) if the hackathon is not approved. However, with this simple approach, a state change of `pending -> approved -> pending -> approved` within the span of a minute will cause two approval emails to be sent.

There may also be gems out there to help with this (search "ruby debounce").

But tbh, in my opinion, debouncing is overkill for this

Reply:
So should we just leave it as is? And where can I find how long it takes to reflect on the frontend?

Reply:
Yeah, let's leave it as is.

The site is built using Next.js's [Incremental Static Regeneration](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration)

- For the homepage, it revalidates every 1 second. So that means it should be updated to date after 1 second + build time (which is about 1 min, 30 secs)
https://github.com/hackclub/hackathons/blob/0ebb3f0a575424a2165da725a45d6e6702881ba3/pages/index.js#L108

- For the region pages, they revalidate every 10 seconds
https://github.com/hackclub/hackathons/blob/0ebb3f0a575424a2165da725a45d6e6702881ba3/pages/[region].js#L115

So it seems like the delay might be around 2 mins then